probe/data.go

```diff
@@ -100,3 +102,36 @@ func LoadDataFromFile(filename string) error {
 
 	return nil
 }
+
+// CleanDataFile keeps the max backup of data file
+func CleanDataFile(filename string, backups int) {
+
+	// if backups is negative value, keep all backup files
+	if backups < 0 {
+		return
+	}
+
+	// get all of the backup files
+	pattern := filename + "-*"
+	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		log.Errorf("Cannot clean data file: %v", err)
+		return
+	}
+
+	// if backups is not exceed the max number of backup files, return
+	if len(matches) <= backups {
+		log.Debugf("No need to clean data file (%d - %d) ", backups, len(matches))
+	}
+
+	// remove the oldest backup files
+	sort.Strings(matches)
```

Comment:
I am not sure how this sorting works, but usually numeric sorting of strings has a few issues, check the example with linux sort command, make sure we dont end up removing a newer file by accident.
```shell
$ seq 1 10|sort
1
10
2
3
4
5
6
7
8
9
```

Reply:
The backup files end with the time RFC3339 format string. So, I think the string sort should be fine. 

```
data.yaml-2022-05-13T17:10:35
data.yaml-2022-05-13T17:13:08 
data.yaml-2022-05-14T15:28:08 
data.yaml-2022-05-14T15:30:39 
data.yaml-2022-05-14T15:34:35
```